5_analyze_data/notebooks/UMAP_analysis/UMAPutils.py

```diff
@@ -0,0 +1,68 @@
+import pathlib
+import pandas as pd
+
+def split_data(pycytominer_output: pd.DataFrame):
+    """
+    split pycytominer output to metadata dataframe and np array of feature values
+    Parameters
+    ----------
+    pycytominer_output : pd.DataFrame
+        dataframe with pycytominer output
+    Returns
+    -------
+    pd.Dataframe, np.ndarray
+        metadata dataframe, feature values
+    """
+    # split metadata from features
+    metadata_cols = [
+        col_name
+        for col_name in pycytominer_output.columns.tolist()
+        if "Metadata" in col_name
+    ]
+    metadata_dataframe = pycytominer_output[metadata_cols]
+
+    feature_cols = [
+        col_name
+        for col_name in pycytominer_output.columns.tolist()
+        if "Metadata" not in col_name
+    ]
+    feature_data = pycytominer_output[feature_cols].values
+
+    return metadata_dataframe, feature_data
+
+
+def merge_metadata_embeddings(
+    metadata_dataframe: pd.DataFrame, embeddings: pd.DataFrame, save_path: pathlib.Path = None
+):
+    """
+    merge metadata with UMAP embeddings into one dataframe
+
+    Parameters
+    ----------
+    metadata_dataframe : pd.Dataframe
+        metadata for the NF1 single cells
+    embeddings : pd.Dataframe
+        2D UMAP embeddings for the x,y coordinates for each single cell
+
+    Returns
+    -------
+    pd.Dataframe
+    """
+    # reset index to remove the 'Metadata_WellRow' as the index then drop the index
+    metadata_dataframe = metadata_dataframe.reset_index()
+    metadata_dataframe = metadata_dataframe.reset_index(drop=True)
```

Comment:
Seeing the repeated calls here makes me wonder if you may only need one. Would it make sense to remove the second `reset_index`?
```suggestion
    # reset index to remove the 'Metadata_WellRow' as the index then drop the index
    metadata_dataframe = metadata_dataframe.reset_index()
```

Reply:
I have to have both as to firstly remove the column of metadata from being the index and then remove the index entirely due to error caused when concatenating the two data frames together.

I wish it could be removed and only have one but I need to reset the index first before removing it.